Update en-scn-connection (#397)
klaytn/klaytn-docs: docs/node/service-chain/getting-started/en-scn-connection.md

> This section covers how to connect your 4-node ServiceChain to the Baobab network.

Author: Thank you. Git pushed...

> You will set up a Baobab EN and connect the EN with one of your SCNs. Then you will enable the Anchoring feature to write ServiceChain block information on Baobab network.

Reviewer: ```suggestion
You will set up a Baobab EN and connect the EN with one of your SCNs. Then you will enable the anchoring feature to put ServiceChain block information onto the Baobab network.
```

Author: Thank you. Git pushed...

![](../images/sc-en-scn-arch.png)

## Prerequisites <a id="prerequisites"></a>
 - 1 Linux or MacOS server for EN
 - Minimum hardware requirements for testing
   - CPU: 4-core (Intel Xeon or equivalent), RAM: 16GB, HDD: 50GB
   - Please refer to the [System Requirements](../references/system-requirements.md) for more explanation.

>  - Download the Baobab EN executable. For the full list of downloadables, see [Download](../../download/README.md).

Reviewer: ```suggestion
 - Download the Baobab EN executable. For the full list of downloadable binaries, see [Download](../../download/README.md).
```

Author: Thank you. Git pushed...

 - Assumptions and Limitations

>    - ServiceChain is installed and running.([link](4nodes-setup-guide.md))

Reviewer: ```suggestion
   - A ServiceChain network is installed and running. Please refer to [Setting up a 4-node Service Chain](4nodes-setup-guide.md) to setup a network.
```

Author: Thank you. Git pushed...

>    - EN connects to Baobab testnet.

Reviewer: ```suggestion
   - A Baobab EN
```

Author: Thank you. Git pushed...

>    - One EN can only have one SCN one-to-one connection.

Reviewer: ```suggestion
   - One EN can only connected to one SCN since only one-to-one connection is supported.
```

Author: Thank you. Git pushed...

>    - Not every SCN has to connect to the EN.

Reviewer: ```suggestion
   - Every SCN does not have to connect to the EN.
```

Author: Thank you. Git pushed...

If you want to stop a node, you can use the command `kend stop`

## Step 5: Check KNI of EN Node <a id="step-5-check-kni-of-en-node"></a>

> Take note of EN-01's KNI which is the information used to connect from an SCN-L2-01 node. This value will be used in the next step when generating `main-bridges.json`

Reviewer: ```suggestion
Take note of EN-01's KNI which is the information used to connect from an SCN-L2-01 node. This value will be used in the next step when generating `main-bridges.json`.
```

Author: Thank you. Git pushed...

```

> $ ken attach --datadir ~/data

Reviewer: ```suggestion
EN-01$ ken attach --datadir ~/data
```

Author: Thank you. Git pushed...

> mainbridge.nodeInfo.kni
"kni://0f7aa6499553...25bae@[::]:50505?discport=0"
```

![](../images/sc-en-scn-nodeInfo.png)

## Step 6: Create main-bridges.json <a id="step-6-create-main-bridges-json"></a>
Log on to an SCN-L2-01 (note: not the EN-01 node) and create `main-bridges.json` on `~/data`. Replace `[::]` located after `@` letter with EN-01 node's IP address.
```

> $ echo '["kni://0f7aa6499553...25bae@192.168.1.1:50505?discport=0"]' > ~/data/main-bridges.json

Reviewer: ```suggestion
SCN-L2-01$ echo '["kni://0f7aa6499553...25bae@192.168.1.1:50505?discport=0"]' > ~/data/main-bridges.json
```

Author: Thank you. Git pushed...

```

> ## Step 7: Configure SCN then Reboot <a id="step-7-configure-scn-then-reboot"></a>

Reviewer: ```suggestion
## Step 7: Configure SCN then Restart kscn <a id="step-7-configure-scn-then-reboot"></a>
```

Author: Thank you. Git pushed...

...
```

> Reboot the SCN-L2-01 node

Author: Thank you. Git pushed...

```

> $ kscnd stop

Reviewer: ```suggestion
SCN-L2-01$ kscnd stop
```

Author: Thank you. Git pushed...

Shutting down kscnd: Killed
$ kscnd start
Starting kscnd: OK
```

Check if the SCN-L2-01 is connected to the EN-01 by checking `subbridge.peers.length`
```
$ kscn attach --datadir ~/data
> subbridge.peers.length
1
```

## Anchoring  <a id="anchoring"></a>
After finishing the EN-01 and SCN-L2-01 connection, you can log ServiceChain block information on the parent chain via Anchoring.
In this section, you will top up a parent operator account, enable Anchoring, and check the anchored block number.

### Step 1: Get KLAY to test anchoring <a id="step-1-get-klay-to-test-anchoring"></a>
To do an anchoring, SCN-L2-01 has to make an anchoring transaction to Baobab. So `subbridge.parentOperator` account should have enough KLAY to pay the transaction fee. Get some KLAY from [Baobab Wallet Faucet](https://baobab.wallet.klaytn.com/) and transfer some KLAY to the `parentOperator`. For data anchoring in real service, `parentOperator` needs to have enough KLAY for transaction fee.
```
$ kscn attach --datadir ~/data
> subbridge.parentOperator
"0x3ce216beeafc62d20547376396e89528e1d778ca"
```

> ![](../images/sc-en-scn-faucet.png)

Reviewer: @nohkwak Where did you get the Faucet image? Does it have any license issue?

Author: I have captured image from the Klaytn wallet fauset.

> ### Step 2: Start Anchoring <a id="step-2-start-anchoring"></a>

Reviewer: ```suggestion
### Step 2: Start Anchoring <a id="step-2-start-anchoring"></a>
To start anchoring, execute the following command:
```

Author: Thank you. Git pushed...

```

## Anchoring  <a id="anchoring"></a>
After finishing the EN and SCN connection, you can log ServiceChain block information on the parent chain via Anchoring.
In this section, you will top up a parent operator account, enable Anchoring, and check the anchored block number.

### Step 1: Get KLAY to test anchoring <a id="step-1-get-klay-to-test-anchoring"></a>

> To do an anchoring, SCN has to make an anchoring transaction to Baobab. So `subbridge.parentOperator` account should have enough KLAY to pay the transaction fee. Get some KLAY from [Baobab Wallet Faucet](https://baobab.wallet.klaytn.com/) and transfer 5 KLAY to the `subbridge.parentOperator`. For data anchoring in real service, parentetOperator needs to have enough KLAY for transaction fee.

Reviewer: ```suggestion
To do an anchoring, SCN has to make an anchoring transaction to Baobab. So `subbridge.parentOperator` account should have enough KLAY to pay the transaction fee. Get some KLAY from [Baobab Wallet Faucet](https://baobab.wallet.klaytn.com/) and transfer some KLAY to the `subbridge.parentOperator`. For data anchoring in real service, parentOperator needs to have enough KLAY for transaction fee.
```
